packages/bpk-component-info-banner/src/BpkInfoBannerInner.tsx

```diff
@@ -0,0 +1,228 @@
+  } as const;
+  const Icon = CustomIcon || componentMap[type];
+  const AlignedIcon = withButtonAlignment(Icon);
+
+  return <AlignedIcon className={className} />;
+};
+
+type ToggleButtonProps = {
+  label?: string;
+  expanded: boolean;
+};
+
+const ToggleButton = (props: ToggleButtonProps) => {
+  const classNames = [getClassName('bpk-info-banner__expand-icon')];
+  if (props.expanded) {
+    classNames.push(getClassName('bpk-info-banner__expand-icon--flipped'));
+  }
+
+  return (
+    <button
+      type="button"
+      className={getClassName('bpk-info-banner__toggle-button')}
+      aria-label={props.label}
+      aria-expanded={props.expanded}
+      title={props.label}
+    >
+      <ExpandIcon className={classNames.join(' ')} />
+    </button>
+  );
+};
+
+type Props = CommonProps & {
+  configuration?: (typeof CONFIGURATION)[keyof typeof CONFIGURATION];
+
+  // Only relevant when configuration == CONFIGURATION.EXPANDABLE
+  children?: ReactNode | string;
+  expanded?: boolean;
+  toggleButtonLabel?: string;
+  onExpandToggle?: OnExpandToggleHandler;
+
+  // Only relevant when configuration == CONFIGURATION.DISMISSABLE
+  dismissButtonLabel?: string;
+  onDismiss?: OnDismissHandler;
+};
+
+const BpkInfoBannerInner = ({
+  animateOnEnter = false,
+  animateOnLeave = false,
+  bannerClassName,
+  children = null,
+  configuration,
+  dismissButtonLabel = '',
+  expanded = false,
+  icon = null,
+  message,
+  onDismiss = null,
+  onExpandToggle = null,
+  show = true,
+  style = STYLE_TYPES.DEFAULT,
+  toggleButtonLabel = '',
+  type = ALERT_TYPES.INFO,
+  ...rest
+}: Props) => {
+  const onBannerExpandToggle = () => {
+    if (onExpandToggle) {
+      onExpandToggle(!expanded);
+    }
+  };
+
+  const onBannerDismiss = () => {
+    if (onDismiss) {
+      onDismiss();
+    }
+  };
+
+  const isExpandable = configuration === CONFIGURATION.EXPANDABLE;
+  const dismissable = configuration === CONFIGURATION.DISMISSABLE;
+  const showChildren = isExpandable && expanded;
+
+  const headerClassNames = [getClassName('bpk-info-banner__header')];
+  const sectionClassNames = [
+    'bpk-info-banner',
+    `bpk-info-banner--${type}`,
+    `bpk-info-banner--style-${style}`,
+  ].map((sectionClassName) => getClassName(sectionClassName));
+
+  if (bannerClassName) {
+    sectionClassNames.push(bannerClassName);
+  }
+
+  if (isExpandable) {
+    headerClassNames.push(getClassName('bpk-info-banner__header--expandable'));
+  }
+
+  /* eslint-disable
+    jsx-a11y/no-static-element-interactions,
+    jsx-a11y/click-events-have-key-events,
+    */
+  // Disabling 'click-events-have-key-events and interactive-supports-focus' because header element is not focusable.
+  // ToggleButton is focusable and works for this.
+  return (
+    <AnimateAndFade
+      animateOnEnter={animateOnEnter}
+      animateOnLeave={dismissable || animateOnLeave}
+      show={show}
+      {...rest}
+    >
+      <section className={sectionClassNames.join(' ')} role="alert">
```

Comment:
If taking above change

```suggestion
      <section className={sectionClassNames} role="alert">
```

Reply:
With above change of sectionClassNames, we'll still have an array of strings instead of string, I assume

Reply:
Not with the above, as the magic that works here is that it will provide out a string of the classNames that are passed to the `getClassNames` function.

So your result would be .e.g `bpk-info-banner-class myCustomProvidedClass`